src/components/Libraries/LibraryListTable.tsx

```diff
@@ -0,0 +1,333 @@
+import { ILibraryMetadata, LibraryIdentifier, useDeleteLibrary } from '@api';
+import {
+  ChevronDownIcon,
+  LockIcon,
+  SettingsIcon,
+  TriangleDownIcon,
+  TriangleUpIcon,
+  UnlockIcon,
+  UpDownIcon,
+} from '@chakra-ui/icons';
+import {
+  Icon,
+  Table,
+  TableProps,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+  Flex,
+  Text,
+  Tooltip,
+  Center,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  useToast,
+  Button,
+  useBreakpoint,
+} from '@chakra-ui/react';
+import { ControlledPaginationControls } from '@components';
+import { CustomInfoMessage } from '@components/Feedbacks';
+import { UserGroupIcon, UserIcon } from '@heroicons/react/24/solid';
+import { NumPerPageType } from '@types';
+import { noop, parseAPIError } from '@utils';
+import { useRouter } from 'next/router';
+import { uniq } from 'ramda';
+import { Fragment, MouseEvent, useMemo } from 'react';
+import { DeleteLibrary } from './DeleteLibrary';
+
+type Column = keyof ILibraryMetadata;
+type SortDirection = 'asc' | 'desc';
+
+const columns: { id: Column; heading: string; sortable: boolean }[] = [
+  {
+    id: 'public',
+    heading: '',
+    sortable: false,
+  },
+  {
+    id: 'num_users',
+    heading: '',
+    sortable: false,
+  },
+  {
+    id: 'name',
+    heading: 'Library',
+    sortable: false, // TODO: true
+  },
+  {
+    id: 'num_documents',
+    heading: 'Papers',
+    sortable: false, // TODO: true
+  },
+  {
+    id: 'owner',
+    heading: 'Owner',
+    sortable: false, // TODO: true
+  },
+  {
+    id: 'permission',
+    heading: 'Permission',
+    sortable: false, // TODO: true
+  },
+  {
+    id: 'date_last_modified',
+    heading: 'Last Modified',
+    sortable: true,
+  },
+];
+
+// hide columns for small display
+const hideColsSmallDisplay: Column[] = ['public', 'num_users', 'permission', 'date_last_modified'];
+
+export interface ILibraryListTableSort {
+  col: keyof ILibraryMetadata;
+  dir: SortDirection;
+}
+
+export interface ILibraryListTableProps extends TableProps {
+  libraries: ILibraryMetadata[];
+  entries: number;
+  sort: ILibraryListTableSort;
+  pageSize: NumPerPageType;
+  pageIndex: number;
+  showIndex?: boolean;
+  showSettings?: boolean;
+  hideCols?: Column[];
+  showDescription?: boolean;
+  onChangeSort: (sort: ILibraryListTableSort) => void;
+  onChangePageIndex: (index: number) => void;
+  onChangePageSize: (size: NumPerPageType) => void;
+  onLibrarySelect: (id: LibraryIdentifier) => void;
+  onUpdate?: () => void;
+}
+
+export const LibraryListTable = (props: ILibraryListTableProps) => {
```

Comment:
Instead of rolling your own table here, you can use the react-table lib I added for the ORCiD works table.  Since it handles sorting/ordering/pagination/etc

Reply:
I was looking at that before, but my table rows are fetched on the fly, and the sorting is a params in the query.